lib/board.rb

```diff
@@ -2,21 +2,84 @@ class Board
 
   def initialize(side_length)
     @side_length = side_length
-    @board = Array.new(@side_length * @side_length, "")
+    @all_moves = set_all_moves(@side_length)
+    @player_moves = Hash.new
+  end
+
+  def to_string_array
+    string_array = Array.new(@all_moves.size, "")
+
+    @player_moves.each do |token, moves_set|
+      assign_moves_to_array(string_array, token, moves_set)
+    end
+
+    string_array
   end
 
   def move(move, player_token)
-    move_index = move - 1
+    if !@player_moves.has_key?(player_token)
+      @player_moves[player_token] = Set.new [move] 
+    else
+      @player_moves[player_token].add(move)
+    end
+  end
+
+  def get_available_moves
+    available_moves_set = @all_moves
 
-    @board[move_index] = player_token
+    @player_moves.each do |token, moves_set|
+      available_moves_set -= moves_set
+    end
+    
+    available_moves_set.to_a
   end
 
-  def to_string_array
-    @board
+  def set_board(board_string_array)
```

Comment:
It seems odd to have a method that replaces nearly the entire state of the object.  It might make sense to convert this into a static method that returns a new object.  It would function kinda like an alternate constructor.

Reply:
I'm not exactly sure how this would help. I would still need to use player_moves to create the string array, since I would like to be able to call the method anytime after the Board is initialized.

Reply:
It's just not good design to pretty much replace the core data (board) of a class. It's better to create a new object if you're going to do that, which is why he suggested an alternate constructor.